src/harness/util.py

```diff
@@ -408,3 +409,69 @@ def filterChannelsByFrequencyRange(channels, freq_range):
       and
       channel['frequencyRange']['highFrequency'] <= freq_range['highFrequency']
   ]
+
+def _orderAttributes(obj):
+    if isinstance(obj, dict):
+        return sorted((k, _orderAttributes(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return sorted(_orderAttributes(x) for x in obj)
+    else:
+        return obj
+
+def compareDictWithUnorderedLists(first_dict, second_dict):
+  """ Deep comparison of two dictionaries
+
+  Args:
+    first_dict: first dictionary to be compared.
+    second_dict: second dictionary to be compared.
+ Returns: boolean set to true if the dictionaries are equal and false otherwise.
+  """
+  return _orderAttributes(first_dict) == _orderAttributes(second_dict)
+
+def isPolygonCCW(polygon):
+  """
+  Args:
+    polygon: the reference GeoJSON geometry of  polygon object.
+
+   Returns: boolean set to true if the coordinates are CCW oriented  and false otherwise.
+  """
+  shapely_polygon = utils.GeoJsonToShapelyGeometry(polygon);
+  print(shapely_polygon.exterior)
+  if not shapely_polygon.exterior.is_ccw:
+    return False
+  #check the holes of the polygons
+  for ring in shapely_polygon.interiors:
+    if not ring.is_ccw:
+      return False
+  return True;
+
+def areTwoPpasEqual(first_ppa, second_ppa, delta = 10):
+  """ Deep comparison of two PPAs considering
+
+  Args:
+    first_ppa: a dictionary contains a reference of PPA data to compare to,
+               the points' order of the ploygon of this PPA is used a reference.
+    second_ppa: a dictionary contains PPA data to be compared.
+    delta: an accepted margin for comparing the polygons of PPAs
+
+ Returns: boolean set to true if the PPAs are equal and false otherwise.
+ Note: the PPA should contain only one GeoJson Feature, this feature should contain only one Polygon
```

Comment:
In fact, this routine supports also MultiPolygon here.
The fact that we only support Polygon is just a requirement on the PPA creation code which will return an exception, if a MultiPolygon is built. But your comparison would continue to work with a geometry being a MultiPolygon (perhaps in a revision of the winnnforum spec), so better to mark it as is